data-prepper-plugins/aggregate-processor/README.md

```diff
@@ -190,6 +191,35 @@ While not necessary, a great way to set up the Aggregate Processor [identificati
         { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "bytes": 3100 }
       ```
 
+### <a name="tail_sampler"></a>
+* `tail_sampler`: The system processes incoming events and determines whether or not they should be allowed based on two criteria. The first criterion is based on whether or not an error condition is present. If any of the aggregated events meet this condition, then all events are allowed to be output. The second criterion is triggered when no error condition is specified or if it is false. In this case, only a subset of the events is allowed to pass through, determined by a probabilistic outcome based on the configured percent value. Since it is difficult to determine exactly when "tail sampling" should occur, the wait_period configuration parameter is used to determine when to conduct this sampling based on the idle time after the last received event. When this action is used, the aggregate `group_duration` is not relevant as the conclusion is based on the `wait_period` and not on the group duration.
+    * It supports the following config options
+       * `percent`: percent of events to be allowed during aggregation window
+       * `wait_period`: minimum idle time before tail sampling is triggered
+       * `error_condition`: optional condition to indicate the error case for tail sampling
+    * When the following three events arrive with `percent` is set to 33, and no error condition specified (or error condition evaluates to false)
+      ```json
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "bytes": 2500 }
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "bytes": 500 }
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "bytes": 1000 }
+      ```
+      The following Events may be allowed, and no event is generated when the group is concluded (Since this is probablistic sampling, exact output is fully deterministic)
+      ```json
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "bytes": 500 }
+      ```
+    * When the following three events arrive with in one second and the `error_condition` is set to `/bytes > 3000`
+      ```json
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "bytes": 2500 }
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "bytes": 500 }
+        { "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "bytes": 3100 }
+      ```
+      The following Events (all) will be allowed, and no event is generated when the group is concluded
```

Comment:
This documentation has two examples that allow all Events. would be nice to have an example where the error condition is not met, as it is not clear to me what the outcome in that situation is.

Reply:
The outcome is not deterministic because we use probabilistic sampling. That's why I used 100%.